### ISSUE TITLE
Complete Portable work

### ISSUE THREAD

**Issue by vbjay** [gitextensions/gitextensions#5823]
- [ ] #2556
- [ ] #5822 
- [ ] Make it so dev can select portable build configuration and lets user debug portable.  
- [ ] Add xml comments to hint dev to not use registry settings.  

Goal.  IsPortable is designed to allow users to put git,  gitextensions, putty, and a diff/merge tool on a thumbdrive and allow user to run a dev system off of a thumbdrive or cloud storage or other portable means.  Take your dev tools with you.

**Comment by spdr870** [comment]
I often download the portable zip instead of using the msi. I would like to keep this option. The current 3RC2 doesn't work for this, because it uses another location for the settings (where it has no write permissions). Maybe an idea to ask the user where to store the settings the first run?

**Comment by RussKie** [comment]
Just FYI - I created a project for tracking all work related to "portability"

**Comment by vbjay** [comment]
@spdr870 extract to a location you can write to like  Desktop/GE or whatever. The whole point of IsPortable is to store settings in folder where GE exists. It needs to be writable

**Comment by vbjay** [comment]
Just fyi. You will lose the canary being self contained and not screwing
with installed ge settings. Think. I run cannary. It has its settings and
my installed has its settings. You can copy your settings file from your
profile to the extracted location if needed.

**Comment by klorinczi** [comment]
I agree the request "Maybe an idea to ask the user where to store the settings the first run?"
This would help my problem, too.

**Comment by vbjay** [comment]
No.  The issue with that is they take the flash drive to another computer and OOPS my settings didn't go with me.  When it is portable, settings are stored in the folder where the exe is.  Period.

**Comment by klorinczi** [comment]
Exactly.

Also note how others do:
Example: Notepad++ has option in installer whether to save settings to %appdata%\notepad folder or not...

**Comment by vbjay** [comment]
The goal of portable is not to allow non install and run.  The goal of portable is to be a self contained portable as in movable from one system to another and it works.  It is a byproduct of this.  To have this portability it needs to be self contained.  So no.  I will not be implementing nag screens asking about do you want to copy or copying settings from an installed( or 2.5 version that basically acted as an installed version because config file was not set correctly.  I have since fixed that issue.)

**Comment by vbjay** [comment]
One of the MANY issues with copying settings from %appdata%.  Copy to another system.  I see all my favorites and recents.  Oops no.  That path doesn't exist.  Like I said it needs to be self contained and it is not quite where we want it.  

Where the idea of portability came from.  I used to tutor students at a community college.  I tried so hard to get them to use git for their work.  I wanted a tooling that they could put on their flash drive and insert into a lab computer or their home computer or save to cloud and then download.  It was their settings their work.  I wanted their settings to go with them and also multiple students could use a lab computer.  That was fun setting user names and email each time you sat down and even when you config your repo, they didn't always remember to do so.

**Comment by klorinczi** [comment]
Being portable or not: 
- it is just a switch which decides the following:
A) load the settings file from current directory,
B) load settings from registry or %appdata% user directory.

**Comment by vbjay** [comment]
Registry is NOT portable. 

![image](https://user-images.githubusercontent.com/3629489/52164814-6e276880-26c5-11e9-9566-f38d6c4c095f.png)


Do you now get it?  You can change the setting to make it not portable yourself.

**Comment by klorinczi** [comment]
Great! 
Then it is already implemented.

But the IsPortable option is only available in registry only?
Can I find it in Gitextensions Settings?

**Comment by vbjay** [comment]
Look again.  Look at which file I selected.

**Comment by klorinczi** [comment]
So if the settings file exists in execution directory, then the app turns into Portable mode?

**Comment by vbjay** [comment]
![image](https://user-images.githubusercontent.com/3629489/52164945-a62fab00-26c7-11e9-9854-3ef4eca65f55.png)

**Comment by vbjay** [comment]
No.  I would think what I posted earlier would tell you how it works.  https://user-images.githubusercontent.com/3629489/52164814-6e276880-26c5-11e9-9566-f38d6c4c095f.png

**Comment by vbjay** [comment]
I can lead you to the file but I can't make you open it.

**Comment by gerhardol** [comment]
Portable should be visible in the About dialog (it is kind of implied from the other changes).

Documentation should describe Portable better too, how to change to/from portable.
https://git-extensions-documentation.readthedocs.io/en/latest/getting_started.html?highlight=portable#portable

**Comment by vbjay** [comment]
See #5780

**Comment by klorinczi** [comment]
What if the registry is set IsPortable=true, but the settings/config file is missing from the app start directory?
What will do? 

What if the registry does not have any stored values? 
By default will load the the settings/config file from from the app start directory?

**Comment by vbjay** [comment]
WHy do you keep saying registry?

**Comment by vbjay** [comment]
I have pointed you to the file.  I would expect you to actually open the file and read it.  It has a comment there to make things quite clear.

**Comment by vbjay** [comment]
No settings file means fresh instance that is ready to configure SEPARATELY from the installed version. 
![image](https://user-images.githubusercontent.com/3629489/52165753-ab462780-26d2-11e9-85d8-009b7b72349b.png)

**Comment by vbjay** [comment]
I have done all but remote into your computer and edit the file myself.  

![image](https://user-images.githubusercontent.com/3629489/52165804-5951d180-26d3-11e9-9491-bf71d4e1519a.png)

**Comment by klorinczi** [comment]
Sorry. I misunderstood one screenshot you posted. 
I thought it was from registry, but it is not.

**Comment by vbjay** [comment]
No that would be xml.
